
  === THIS PATCH REVERTS:
  === http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=7ad1227818f09242cfe9bf1845fd24211f5f99bd


From 7ad1227818f09242cfe9bf1845fd24211f5f99bd Mon Sep 17 00:00:00 2001
From: Jan Beulich <JBeulich@novell.com>
Date: Thu, 9 Dec 2010 08:11:38 +0000
Subject: [PATCH] kconfig: fix undesirable side effect of adding "visible"
 menu attribute

This lead to non-selected, non-user-selectable options to be written
out to .config. This is not only pointless, but also preventing the
user to be prompted should any of those options eventually become
visible (e.g. by de-selecting the *_AUTO options the "visible"
attribute was added for.

Furthermore it is quite logical for the "visible" attribute of a menu
to control the visibility of all contained prompts, which is what the
patch does.

Signed-off-by: Jan Beulich <jbeulich@novell.com>
Signed-off-by: Michal Marek <mmarek@suse.cz>
---
 scripts/kconfig/menu.c |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

diff --git scripts/kconfig/menu.c scripts/kconfig/menu.c
index b9d9aa1..5f77dcb 100644
--- scripts/kconfig/menu.c
+++ scripts/kconfig/menu.c
@@ -141,6 +141,7 @@
 		if (current_entry->prompt && current_entry != &rootmenu)
 			prop_warn(prop, "prompt redefined");
 
+#if 0
 		/* Apply all upper menus' visibilities to actual prompts. */
 		if(type == P_PROMPT) {
 			struct menu *menu = current_entry;
@@ -166,6 +167,7 @@
 							 dup_expr);
 			}
 		}
+#endif
 
 		current_entry->prompt = prop;
 	}
